packages/core/src/integrations/supabase.ts

```diff
@@ -215,6 +217,76 @@ export function translateFiltersIntoMethods(key: string, query: string): string
   return `${method}(${key}, ${value.join('.')})`;
 }
 
+function instrumentRpcReturnedFromSchemaCall(SupabaseClient: unknown): void {
+  (SupabaseClient as unknown as SupabaseClientConstructor).prototype.schema = new Proxy(
+    (SupabaseClient as unknown as SupabaseClientConstructor).prototype.schema,
+    {
+      apply(target, thisArg, argumentsList) {
+        const rv = Reflect.apply(target, thisArg, argumentsList);
+
+        return instrumentRpc(rv);
+      },
+    },
+  );
+}
+
+function instrumentRpc(SupabaseClient: unknown): unknown {
+  (SupabaseClient as unknown as SupabaseClientConstructor).rpc = new Proxy(
+    (SupabaseClient as unknown as SupabaseClientConstructor).rpc,
+    {
+      apply(target, thisArg, argumentsList) {
+        const isProducerSpan = argumentsList[0] === 'enqueue';
+        const isConsumerSpan = argumentsList[0] === 'dequeue';
```

Comment:
I don't know if this recently changed, but here they show `send` and `pop` as rpc args: https://supabase.com/docs/guides/queues/quickstart#enqueueing-and-dequeueing-messages :thinking: